src/assignment-cache.ts

```diff
@@ -0,0 +1,78 @@
+import { LRUCache } from 'lru-cache';
+
+import { EppoValue } from './eppo_value';
+
+export interface AssignmentCacheKey {
+  subjectKey: string;
+  flagKey: string;
+  allocationKey: string;
+  variationValue: EppoValue;
+}
+
+interface Cacheable {
+  get(key: string): string | undefined;
+  set(key: string, value: string): void;
+  has(key: string): boolean;
+}
+
+export type AvailableCacheTypes = Map<string, string> | LRUCache<string, string>;
```

Comment:
The problem I was trying to solve with this refactor into an abstract class was that the implementations for `hasLoggedAssignment` and `logAssignment` were identical, save for the `cache` parameter.

Feels like I got a bit lucky that the `Map` and `LRUCache` interfaces happen to both implement `Cacheable` the exact same way. 

A bit of a down-side with this type is not allowing users to provide their own implementation beyond `Map` and `LRUCache`.

Do you think this is going too far with the refactor?

Reply:
Why further restrict? Does just having `Cacheable` as the type not work? Then you could have a third function like `setCache()`.

(Sorry if dumb question, I'm not the most knowledgeable with TypeScript inheritance)

Although, the more I think about it, I'd be inclined to keep these two options until a customer asks to use a custom one that doesn't extend one of these two 🤷‍♂️ So I say carry on.